### ISSUE TITLE
Image widget size only dependent on width

### ISSUE THREAD

**Issue by Shootertrex** [hecrj/iced#110]
I think there's actually 2 connected issues here. 

First is that no matter what configuration I set for the image, it will always fill `width`, even if `width` is set to `Length::Shrink` and `height` is set to `Length::Fill`.  

Second is that the image cannot be bound within a container. By this I mean that if the image is taller than the container is tall, it will overflow out of the canvas instead of shrinking the height. Here is the code snippet for the the image and the screenshots to show what I mean:
```Rust
Container::new(
    Image::new("/home/user/Pictures/tall.png")
        .width(Length::Shrink)
        .height(Length::Fill)
)
    .width(Length::Fill)
    .center_x()
```
Image being shrunk by narrow window: 
![2019-12-06_16:23:14_364x630](https://user-images.githubusercontent.com/6365966/70357836-e3178180-1845-11ea-8fdb-9842ab2fbb26.png)

Image being cut off in a wide window:
![2019-12-06_16:22:40_839x419](https://user-images.githubusercontent.com/6365966/70357866-f88cab80-1845-11ea-8aa8-9ca833a8835c.png)

I see that there is a TODO in the Image widget's `layout` method, so this could just be functionality not yet implemented. A possible solution (albeit maybe a naive one) is to modify that method to look something like this:
```Rust
let mut size = limits.width(width).height(height).resolve(Size::ZERO);

size.height = size.width / aspect_ratio

// new code
if size.height > limits.max().height {
    size.height = limits.max().height;
    size.width = size.height / (1 as f32 /aspect_ratio);
}
// end new code

layout::Node::new(size)
```

This solution, while fixes it for my use case (having the entire image visible at all times), it removes the ability to have the image flow off the screen, if that is at all desired. I'm interested to hear your thoughts on this.

**Comment by hecrj** [comment]
Hey @Shootertrex,

I think this may have been fixed by #90 here: https://github.com/hecrj/iced/blob/1f60e2820463ef9f7fe9dda5d8b8fa040fdd8666/native/src/widget/image.rs#L70-L93

Have you tried it with the latest `master` changes? For instance, the `pokedex` example shows how the `Image` always stays visible:

![image](https://user-images.githubusercontent.com/518289/70359376-0f4df500-187c-11ea-8b3a-146a42f08032.png)

**Comment by Shootertrex** [comment]
Thanks for the quick response. You are correct that #90 fixed it. I had ran into this earlier in the week, but I wasn't expecting it to have already been fixed!

I'll close this issue then.
